install.lua

```diff
@@ -0,0 +1,66 @@
+-- Easy Installer for computercraft-github by Eric Wieser
+-- https://github.com/eric-wieser/computercraft-github
+
+local tree = select(1,...)
+if not tree then
+  tree = 'master'
+end
+
+local REPO_BASE = ('https://raw.githubusercontent.com/eric-wieser/computercraft-github/%s/'):format(tree)
+
+local FILES = {
+  'apis/dkjson',
+  'apis/github',
+  'programs/github',
+  'github'
+}
+
+local function request(url_path)
+  local request = http.get(REPO_BASE..url_path)
+  status = request.getResponseCode()
+  response = request.readAll()
+  request.close()
+  return status, response
+end
+
+local function makeFile(file_path, data)
+ local file = fs.open('github.rom/'..file_path,'w')
+ file.write(data)
+ file.close()
+end
+
+local function rewriteDofiles()
+  for _, file in pairs(files) do
+    local filename = ('github.rom/%s'):format(file)
+    local r = fs.open(filename, 'r')
+    local data = r.readAll()
+    r.close()
+    local w = fs.open(filename, 'w')
+    data = data.gsub('dofile("', 'dofile("github.rom/')
```

Comment:
yeah I found it can you merge my fixes into pr-4 please


On Thu, Aug 25, 2016 at 11:33 PM Eric Wieser <notifications@github.com>
wrote:

> In install.lua
> <https://github.com/eric-wieser/computercraft-github/pull/4#discussion_r76363125>
> :
>
> > +end
> > +
> > +local function makeFile(file_path, data)
> > + local file = fs.open('github.rom/'..file_
>
> path,'w')
> > + file.write(data)
> > + file.close()
> > +end
> > +
> > +local function rewriteDofiles()
> > +  for _, file in pairs(files) do
> > +    local filename = ('github.rom/%s'):format(file)
> > +    local r = fs.open(filename, 'r')
> > +    local data = r.readAll()
> > +    r.close()
> > +    local w = fs.open(filename, 'w')
> > +    data = data.gsub('dofile("', 'dofile("github.rom/')
>
> From the documentation,
>
> Parentheses have a special meaning in a pattern, so we must escape them
> with a %
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/eric-wieser/computercraft-github/pull/4/files/86a13b34c4ce8f00c62e5a028fcfb8ba6d233427#r76363125>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA41FiSNfxa_lClAO4wjBoWWWCRWx9Teks5qjmx7gaJpZM4JoHt0>
> .
>